drivers: spi: Add support Renesas SCI_B_SPI

khoa-nguyen-18 opened the pull request:
* Add support for Renesas `spi_renesas_ra_sci_b` driver
* Enable support on Renesas `ek_ra8m1`, `ek_ra8d1`, and `mck_ra8t1` boards
* Add support for `spi` test apps using the `spi_renesas_ra_sci_b` driver on `ek_ra8m1`, `ek_ra8d1`, and `mck_ra8t1`
* Add support for `disk` and `sdhc` tests and sample apps using the `spi_renesas_ra_sci_b` driver on `ek_ra8m1`, `ek_ra8d1`, and `mck_ra8t1`

khoatranyj committed:
drivers: spi: Initial driver support for SCI B SPI on Renesas RA

Add SCI B SPI driver support for Renesas RA

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
dts: arm: renesas: Add SPI device node to SCI device node

for Renesas RA family

Add SPI device node to support SCI B SPI driver on Renesas RA SoCs:
- ra8x1.dtsi
- ra8x2.dtsi

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: disk: Add tests support for SCI B SPI driver

Add test support using SDMMC card on Renesas RA boards

Add Renesas RA board support for these tests:
- tests/drivers/disk/disk_access

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: sdhc: Add tests support for SCI B SPI driver

Add test support using SDMMC card on Renesas RA boards

Add Renesas RA board support for these tests:
- tests/drivers/sdhc

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoa-nguyen-18 commented:
- Rebase to solve the conflicts
- Update based on feedback comments

khoa-nguyen-18 commented on drivers/spi/spi_renesas_ra_sci_b.c:
```diff
@@ -0,0 +1,707 @@
+			} else {
+				/* Do nothing */
+			}
+		}
```
Thank you. We've updated it

quytranpzz commented on drivers/spi/spi_renesas_ra_sci_b.c:
```diff
@@ -0,0 +1,710 @@
+extern void sci_b_spi_eri_isr(void);
+#endif
+
+#define GPIO_ACTIVE_STATE_LOW    BIT(0)
```
Use `GPIO_ACTIVE_LOW`?

khoatranyj committed:
tests: drivers: spi: Add tests support for SCI B SPI on RA devices

Add Renesas RA board support for these tests:
- tests/drivers/spi/spi_loopback
- tests/drivers/spi/spi_controller_peripheral

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
samples: subsys: fs: Add tests support for SCI B SPI

driver using SDMMC card on Renesas RA boards

Add Renesas RA board support for these samples:
- samples/subsys/fs/fs_sample

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

tbursztyka reviewed (commented):
rest lgtm

JarmouniA commented on drivers/spi/spi_renesas_ra_sci_b.c:
```diff
@@ -22,7 +22,8 @@
 #include <zephyr/logging/log.h>
-LOG_MODULE_REGISTER(renesas_ra_spi_sci_b);
+
+LOG_MODULE_REGISTER(renesas_ra_spi_sci_b, CONFIG_SPI_LOG_LEVEL);
```
This should be in the driver's commit

JarmouniA commented on drivers/spi/spi_renesas_ra_sci_b.c:
```diff
@@ -209,8 +210,6 @@ static void spi_renesas_ra_sci_b_callback(spi_callback_args_t *p_args)
 			} else if (data->ctx.tx_buf == NULL) {
 				data->ctx.recv_frames = data->data_len;
-			} else {
-				/* Do nothing */
```
ditto

JarmouniA commented on drivers/spi/spi_renesas_ra_sci_b.c:
```diff
@@ -535,7 +534,8 @@ static DEVICE_API(spi, spi_renesas_ra_sci_b_driver_api) = {
 #endif /* CONFIG_SPI_ASYNC */
-	.release = spi_renesas_ra_sci_b_release};
+	.release = spi_renesas_ra_sci_b_release,
+};
```
ditto

JarmouniA commented on tests/drivers/spi/spi_loopback/testcase.yaml:
```diff
@@ -350,3 +350,41 @@ tests:
+      - CONFIG_SPI_ASYNC=n
+    extra_args:
+      - DTC_OVERLAY_FILE="boards/${BOARD}_sci_b_spi.overlay"
+      - CONF_FILE="prj.conf"
```
```suggestion
```
Will be picked by default.

JarmouniA commented on drivers/spi/Kconfig.renesas_ra:
```diff
@@ -33,3 +33,31 @@ config SPI_USE_HW_SS
+	depends on DT_HAS_RENESAS_RA_SPI_SCI_B_ENABLED
+	select USE_RA_FSP_SCI_B_SPI
+	select PINCTRL
+	select SPI_RENESAS_RA_SCI_B_INTERRUPT if SPI_ASYNC
```
```suggestion
```

JarmouniA commented on drivers/spi/Kconfig.renesas_ra:
```diff
@@ -33,3 +33,31 @@ config SPI_USE_HW_SS
+	  Enable Interrupt support for the SCI B SPI Driver of RA family.
+
+config SPI_RENESAS_RA_SCI_B_DTC
+	bool "RA MCU SCI B SPI DTC Support"
```
```suggestion
	bool
```

JarmouniA commented on drivers/spi/Kconfig.renesas_ra:
```diff
@@ -33,3 +33,31 @@ config SPI_USE_HW_SS
+if SPI_RENESAS_RA_SCI_B
+
+config SPI_RENESAS_RA_SCI_B_INTERRUPT
+	bool "RENESAS RA SCI B SPI Interrupt Support"
```
```suggestion
	bool
	default y
```

khoa-nguyen-18 committed:
boards: renesas: Update supported tag for integration_platforms

Update supported spi and sdhc tag for integration_platforms for
spi and disk test: ek_ra8m1, ek_ra8p1 cm85

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

JarmouniA reviewed (dismissed):
FYI pushing changes requested in review mixed with a rebase on main is not very helpful for reviewers.
https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-tips

khoatranyj committed:
samples: subsys: fs: Add sample support for sdhc on Renesas RA

Add sample support for sdhc driver on Renesas ek_ra8m1 and
ek_ra8p1 board:
- samples/subsys/fs/fs_sample

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: disk: Add test support for sdhc on Renesas RA

Add test support for sdhc driver on Renesas ek_ra8m1 and
ek_ra8p1 board:
- tests/drivers/disk/disk_access

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

thaoluonguw reviewed (approved):
LGTM.
